components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -14704,6 +14706,80 @@ public String getSharedScopeKeyByUUID(String uuid) throws APIManagementException
         return scopeKey;
     }
 
+    /***
+     * Get the API and URI usages of the given shared scope
+     *
+     * @param uuid Id of the shared scope
+     * @param tenantId tenant Id
+     * @return usgaes ofr the shaerd scope
+     * @throws APIManagementException If an error occurs while getting the usage details
+     */
+    public SharedScopeUsage getSharedScopeUsage(String uuid, int tenantId) throws APIManagementException {
+        Connection conn = null;
+        ResultSet apiUsageResultSet = null;
+        ResultSet uriUsageResultSet = null;
+        PreparedStatement psForApiUsage = null;
+        PreparedStatement psForUriUsage = null;
+        SharedScopeUsage sharedScopeUsage = null;
+        List<API> usedApiList = new ArrayList<>();
+        String sharedScopeName = getSharedScopeKeyByUUID(uuid);
+
+        if (sharedScopeName != null) {
+            sharedScopeUsage = new SharedScopeUsage();
+            sharedScopeUsage.setId(uuid);
+            sharedScopeUsage.setName(sharedScopeName);
+        } else {
+            throw new APIMgtResourceNotFoundException("Shared Scope not found for scope ID: " + uuid,
+                    ExceptionCodes.from(ExceptionCodes.SHARED_SCOPE_NOT_FOUND, uuid));
+        }
+
+        try {
+            conn = APIMgtDBUtil.getConnection();
+            String sqlQueryForApiUsage = SQLConstants.GET_SHARED_SCOPE_API_USAGE_BY_TENANT;
+            psForApiUsage = conn.prepareStatement(sqlQueryForApiUsage);
+            psForApiUsage.setString(1, uuid);
+            psForApiUsage.setInt(2, tenantId);
+            apiUsageResultSet = psForApiUsage.executeQuery();
+            while (apiUsageResultSet.next()) {
```

Comment:
is it possible to use try with resources

Reply:
Fixed with 3550c6ac829d5e756dab682eab507f16e5616d0b